test/metabase/query_processor_test/alternative_date_test.clj

```diff
@@ -215,16 +229,33 @@
                  (mt/rows (mt/dataset string-times
                             (qp/process-query
                               (assoc (mt/mbql-query times)
-                                     :middleware {:format-rows? false})))))))))
+                                     :middleware {:format-rows? false}))))))))
+
+     (testing "mongo only supports datetime"
+       (mt/test-drivers #{:mongo}
+         (is (= [[1 "foo" (t/instant "2004-10-19T10:23:54Z")]
+                 [2 "bar" (t/instant "2012-10-19T10:23:54Z")]
+                 [3 "baz" (t/instant "2008-10-19T00:00:00Z")]]
+                (mt/rows (mt/dataset mongo-dates
+                                     (qp/process-query
+                                      (assoc (mt/mbql-query dates)
+                                             :middleware {:format-rows? false})))))))))
+
     (testing "are queryable as dates"
-      (testing "a datetime field"
+      (testing "a datetime field - mongo driver"
+        (mt/test-drivers #{:mongo}
+          (is (= 1
+                 (count (mt/rows (mt/dataset mongo-dates
+                                   (mt/run-mbql-query dates
+                                     {:filter [:= $ts "2008-10-19"]}))))))))
+      (testing "a datetime field - SQL drivers"
         ;; TODO: why does this fail on oracle? gives a NPE
         (mt/test-drivers (disj (sql-jdbc.tu/sql-jdbc-drivers) :oracle :sparksql)
-          (is (= 1
-                 (count (mt/rows (mt/dataset string-times
-                                   (mt/run-mbql-query times
-                                     {:filter   [:= [:datetime-field $ts :day] "2008-10-19"]}))))))))
-      (testing "a date field"
+         (is (= 1
+                (count (mt/rows (mt/dataset string-times
+                                  (mt/run-mbql-query times
+                                    {:filter   [:= [:datetime-field $ts :day] "2008-10-19"]}))))))))
```

Comment:
I'm confused, why doesn't this work for MongoDB. Why do we need a separate test

Reply:
it was failing for Mongo because mong doesn't support casting for datetime and time, and this test is selecting all columns which contain those casting.

I create another similar test that only select the `ts` column for this.